lms/djangoapps/grades/api/v1/tests/test_gradebook_views.py

```diff
@@ -1204,3 +1210,129 @@ def test_override_is_created(self, login_method):
+@ddt.ddt
+class SubsectionGradeViewTest(GradebookViewTestBase):
+    """ Test for the audit api call """
+    @classmethod
+    def setUpClass(cls):
+        super(SubsectionGradeViewTest, cls).setUpClass()
+        cls.namespaced_url = 'grades_api:v1:course_grade_overrides'
+        cls.locator_a = BlockUsageLocator(
+            course_key=cls.course_key,
+            block_type='problem',
+            block_id='block_id_a'
+        )
+        cls.locator_b = BlockUsageLocator(
+            course_key=cls.course_key,
+            block_type='problem',
+            block_id='block_id_b'
+        )
+        cls.record_a = BlockRecord(locator=cls.locator_a, weight=1, raw_possible=10, graded=False)
+        cls.record_b = BlockRecord(locator=cls.locator_b, weight=1, raw_possible=10, graded=True)
+        cls.block_records = BlockRecordList([cls.record_a, cls.record_b], cls.course_key)
+        cls.usage_key = cls.subsections[cls.chapter_1.location][0].location
+        cls.user_id = 12345
+        cls.params = {
+            "user_id": cls.user_id,
+            "usage_key": cls.usage_key,
+            "course_version": "deadbeef",
+            "subtree_edited_timestamp": "2016-08-01 18:53:24.354741Z",
+            "earned_all": 6.0,
+            "possible_all": 12.0,
+            "earned_graded": 6.0,
+            "possible_graded": 8.0,
+            "visible_blocks": cls.block_records,
+            "first_attempted": datetime(2000, 1, 1, 12, 30, 45, tzinfo=UTC),
+        }
+        cls.grade = PersistentSubsectionGrade.update_or_create_grade(**cls.params)
+
+    def get_url(self, subsection_id=None, user_id=None):
+        """
+        Helper function to create the course gradebook API url.
+        """
+        base_url = reverse(
+            self.namespaced_url,
+            kwargs={
+                'subsection_id': subsection_id or self.subsection_id,
+            }
+        )
+        return "{0}?user_id={1}".format(base_url, user_id or self.user_id)
+
+    @ddt.data(
+        'login_staff',
+        'login_course_admin',
+        'login_course_staff',
+    )
+    def test_no_override(self, login_method):
+        getattr(self, login_method)()
+
+        resp = self.client.get(
+            self.get_url(subsection_id=self.usage_key)
+        )
+
+        expected_data = {
+            'original_grade': OrderedDict([
+                ('earned_all', 6.0),
+                ('possible_all', 12.0),
+                ('earned_graded', 6.0),
+                ('possible_graded', 8.0)
+            ]),
+            'user_id': 12345,
+            'override': None,
+            'course_id': text_type(self.course_key),
+            'subsection_id': text_type(self.usage_key),
+            'history': []
+        }
+
+        self.assertEqual(expected_data, resp.data)
+
+    @ddt.data(
+        'login_staff',
+        'login_course_admin',
+        'login_course_staff',
+    )
+    @freeze_time('2019-01-01')
+    def test_with_override(self, login_method):
```

Comment:
Can we have another test case for when override exists, but not history? This case is still valid for we have many overrides out there without history, because history is a recent table and we couldn't back-populate.